### ISSUE TITLE
Clean up exp/views/responses.py

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#517]
**TL;DR**: Clean up the code for generating response files (csv/json summaries, etc.)

**Narrative**
It will be easier to reason about and build on the response views if the code is better structured. The tasks below will also speed up page loading and allow downloads of files with larger or more responses.

**Acceptance Criteria**
- [x] Response/JSON CSVs generated from common definition of fields; optional & identifiable fields extracted based on definitions
- [x] Change response download views to listviews on Response model; use mixin rather than extending StudyResponsesAll
- [x] Use a streaming response for the CSVs to deal with large files better
- [x] Automatically split up large files (>65k lines) and/or don't provide the "all files" frame data, which is huge in most cases, in favor of just the single-file-per-session archive. May need to also defer download of the archive or figure out if we can stream it.
- [x] Similar steps for demographic download: define fields centrally, use mixin rather than extending StudyDemographics, use list views for download views
- [x] Change the individual responses page to avoid loading each format into the DOM. Display instead a small summary (e.g. CSV summary fields) and provide a button to download the selected format. 
- [x] Rico signoff
